pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -686,6 +687,48 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		}
 	}
 
+	// Ensure the install config matches the ClusterDeployment:
+	icSecret := &corev1.Secret{}
+	err = r.Get(context.Background(),
+		types.NamespacedName{
+			Namespace: cd.Namespace,
+			Name:      cd.Spec.Provisioning.InstallConfigSecretRef.Name,
+		},
+		icSecret)
+	if err != nil {
+		cdLog.WithError(err).Log(controllerutils.LogLevel(err), "Error loading install config secret")
+		return reconcile.Result{}, err
+	}
+	err = ValidateInstallConfig(cd, icSecret.Data["install-config.yaml"])
+	if err != nil {
+		cdLog.WithError(err).Info("install config validation failed")
+		conditions, changed := controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+			cd.Status.Conditions,
+			hivev1.ReadyToProvisionCondition,
+			corev1.ConditionFalse,
+			"InstallConfigValidationFailed",
+			err.Error(),
+			controllerutils.UpdateConditionIfReasonOrMessageChange)
+		if changed {
+			cd.Status.Conditions = conditions
+			return reconcile.Result{}, r.Status().Update(context.TODO(), cd)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// If we made it this far, ReadyToProvision condition should be True:
+	conditions, changed := controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+		cd.Status.Conditions,
+		hivev1.ReadyToProvisionCondition,
+		corev1.ConditionTrue,
+		"ProvisionChecksPassed",
+		"Ready to provision",
+		controllerutils.UpdateConditionIfReasonOrMessageChange)
+	if changed {
+		cd.Status.Conditions = conditions
+		return reconcile.Result{}, r.Status().Update(context.TODO(), cd)
+	}
```

Comment:
@suhanime hoping to use this condition to consolidate several of the "pre-flight" conditions we set today prior to launching the install, what do you think? (goal being to trim down the number)

Reply:
Actually now that I think of it, we hope to soon create a ClusterInstall implementation for our current OpenShift installer provisioning. When we do, the contract requires that we would be copying back a ClusterInstallRequirementsMet condition. Probably best to just use this same condtion name here for consistency.

Reply:
Is it really necessary to introduce a new condition? Can't we just use the `ProvisionFailed` condition? We can set the reason+message of what you send from `ValidateInstallConfig`
We already do this with `Hibernating` condition - we explicitly set it to false when a cluster cannot be hibernated without even trying to hibernate it. In my opinion, same logic applies here
Additional benefit - we would get the alert via `ProvisionUnderway` metric so it could always be monitored

Reply:
fwiw, if you do decide to keep this condition around, it becomes a necessary condition for provision to take place, so for the failing tests, if you add this to [defaultConditions](https://github.com/openshift/hive/blob/4376099acc9c78577b1609347655c98bc98def28/pkg/controller/clusterdeployment/clusterdeployment_controller_test.go#L2330-L2347), it might help with some of the failing tests

Reply:
This is the condition we already have [ClusterInstallRequirementsMetClusterDeploymentCondition](https://github.com/openshift/hive/blob/master/apis/hive/v1/clusterdeployment_types.go#L409)
It would be too confusing to have both - one with `ClusterDeployment` in the name and one without...
As far as your idea of using the same one for this purpose... I'm a little bit cautious, but worth looking into

Reply:
Thoughts on conditions: https://github.com/openshift/hive/pull/1413